lib/routes/users-request-reset.js

```diff
@@ -27,20 +28,28 @@ module.exports = (server, options) => {
         handler: async (request) => {
 
             const Users = request.models().Users;
+            const { emailService } = request.services();
             const Payload = request.payload;
 
             const rawResetToken = Buffer.from(Uuid({ rng: Uuid.nodeRNG }));
 
             const hash = Pwd.hashSync(rawResetToken);
 
-            await Users.query()
+            const patchedUser = await Users.query()
                 .patch({ 'password': null, resetToken: hash.toString('utf8') })
-                .where({ 'email': Payload.email });
-            //send token to user via email here. This functionality
-            //intentionally left out, as it is dependent on
-            //infrastructure and front end implementation.
+                .where({ 'email': Payload.email })
+                .first()
+                .returning('*');
 
-            return 'Raw Reset Token, for testing only *' + rawResetToken + '* TODO: Implement email sending';
+            if (patchedUser) {
+
+                const resetUrl = options.siteUrl + '/password-reset/' + rawResetToken;
```

Comment:
Do we care about enforcing that `options.siteUrl` exists? If I use `user-pal` as a dep and forget to add it I won't know til the link gets sent out

Reply:
If that's desired, validating plugin options would make sense to do as "custom code" in `lib/index.js`.

Reply:
I like this idea - there are 2 things from the .env file that are needed at runtime (jwtKey and siteUrl) - i'm going to add some code checking for those to `lib/index.js`